Update CEL documentation to capture type checking errors. (#14107)
istio/istio.io: content/en/docs/tasks/observability/logs/telemetry-api/index.md

    EOF
    {{< /text >}}

1. Filter health check access logs with CEL expression

>     The following configuration displays access logs only when the logs are not generated by the Amazon Route53 Health Check Service.

Reviewer: ```suggestion
    The following configuration displays access logs only when the logs are not generated by the Amazon Route 53 Health Check Service.
```

That will fix the final lint error.

    EOF
    {{< /text >}}

1. Filter health check access logs with CEL expression

>     The following configuration displays access logs only when the logs aren't generated by the Amazon-Route53-Health-Check-Service. 

Reviewer 2: ```suggestion
    The following configuration displays access logs only when the logs aren't generated by the Amazon-Route53-Health-Check-Service.
```

    Note: The `request.useragent` is specific to HTTP traffic, therefore to avoid breaking TCP traffic, we need to check for the existence of the field.
    For more information, see [CEL Type Checking](https://kubernetes.io/docs/reference/using-api/cel/#type-checking)

    {{< text bash >}}
    $ cat <<EOF | kubectl apply -f -
    apiVersion: telemetry.istio.io/v1alpha1
    kind: Telemetry
    metadata:
      name: filter-health-check-logging
    spec:
      accessLogging:
      - providers:
        - name: otel
        filter:

>           expression: "!has(request.useragent) || !(request.useragent.startsWith("Amazon-Route53-Health-Check-Service"))"
> 
>     EOF

Reviewer 2: ```suggestion
          expression: "!has(request.useragent) || !(request.useragent.startsWith("Amazon-Route53-Health-Check-Service"))"
    EOF
```
